dali/util/shared_mem.cc

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <errno.h>
+#include <fcntl.h> /* For O_* constants */
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <exception>
+#include <cstring>
+
+#include "dali/util/shared_mem.h"
+
+namespace dali {
+
+namespace detail {
+
+bool dir_exists(const char * str) {
+  struct stat sb;
+  if (stat(str, &sb) == -1) {
+    return false;
+  }
+  return S_ISDIR(sb.st_mode);
+}
+
+}  // namespace detail
+
+namespace python {
+ShmFdWrapper::ShmFdWrapper(int fd) : fd_{fd} {}
+
+ShmFdWrapper::ShmFdWrapper() {
+  // Abstract away the fact that shm_open requires filename.
+  constexpr char dev_shm_path[] = "/dev/shm/";
+  constexpr char run_shm_path[] = "/run/shm/";
+  constexpr char temp_filename_template[] = "nvidia_dali_XXXXXX";
+  constexpr int kShmPathSize = sizeof(dev_shm_path) - 1;
```

Comment:
```suggestion
  constexpr int kShmPathSize = std::max(sizeof(dev_shm_path), sizeof(run_shm_path)) - 1;
```
Better be safe than sorry.

Reply:
Done.